pkg/controller/jenkins/configuration/base/reconcile.go

```diff
@@ -609,23 +610,23 @@ func (r *ReconcileJenkinsBaseConfiguration) compareContainers(expected corev1.Co
 		messages = append(messages, "Lifecycle has changed")
 		verbose = append(verbose, fmt.Sprintf("Lifecycle has changed to '%+v' in container '%s'", expected.Lifecycle, expected.Name))
 	}
-	if !reflect.DeepEqual(expected.LivenessProbe, actual.LivenessProbe) {
+	if !reflect.DeepEqual(expected.LivenessProbe, actual.LivenessProbe) && !r.isOpenshiftEnabled(){
 		messages = append(messages, "Liveness probe has changed")
 		verbose = append(verbose, fmt.Sprintf("Liveness probe has changed to '%+v' in container '%s'", expected.LivenessProbe, expected.Name))
 	}
 	if !reflect.DeepEqual(expected.Ports, actual.Ports) {
 		messages = append(messages, "Ports have changed")
 		verbose = append(verbose, fmt.Sprintf("Ports have changed to '%+v' in container '%s'", expected.Ports, expected.Name))
 	}
-	if !reflect.DeepEqual(expected.ReadinessProbe, actual.ReadinessProbe) {
+	if !reflect.DeepEqual(expected.ReadinessProbe, actual.ReadinessProbe) && !r.isOpenshiftEnabled(){
 		messages = append(messages, "Readiness probe has changed")
 		verbose = append(verbose, fmt.Sprintf("Readiness probe has changed to '%+v' in container '%s'", expected.ReadinessProbe, expected.Name))
 	}
 	if !reflect.DeepEqual(expected.Resources, actual.Resources) {
 		messages = append(messages, "Resources have changed")
 		verbose = append(verbose, fmt.Sprintf("Resources have changed to '%+v' in container '%s'", expected.Resources, expected.Name))
 	}
-	if !reflect.DeepEqual(expected.SecurityContext, actual.SecurityContext) {
+	if !reflect.DeepEqual(expected.SecurityContext, actual.SecurityContext) && !r.isOpenshiftEnabled() {
 		messages = append(messages, "Security context has changed")
 		verbose = append(verbose, fmt.Sprintf("Security context has changed to '%+v' in container '%s'", expected.SecurityContext, expected.Name))
 	}
```

Comment:
Add `!r.isOpenshiftEnabled()` at beginning and then check the other conditions.
Something like this:

```
if !r.isOpenshiftEnabled() {
    if !reflect.DeepEqual(expected.LivenessProbe, actual.LivenessProbe)...
    if !reflect.DeepEqual(expected.ReadinessProbe, actual.ReadinessProbe)...
    if !reflect.DeepEqual(expected.SecurityContext, actual.SecurityContext)...

   [...]
}

```

Reply:
This doesn't make sense, as I suppose while checking this stuff you guys must ahve kept the order in mind, for what resources are being checked, if not I can do it this way.